### PR TITLE
Add query priority option INTERACTIVE or BATCH

### PR DESCRIPTION
**Pull request by LiorRacer**
**What this PR does / why we need it**:
You can now choose query priority per datasource

image

Seoul location was added
**Which issue(s) this PR fixes**:
https://github.com/doitintl/bigquery-grafana/issues/253
https://github.com/doitintl/bigquery-grafana/issues/256

### CODE REVIEW

File: src/config_ctrl.ts [hunk 1]
```diff
@@ -36,6 +37,10 @@ export class BigQueryConfigCtrl {
     if (this.current.jsonData.processingLocations === undefined) {
       this.current.jsonData.processingLocations = this.defaultProcessingLocation;
     }
+    if (this.current.jsonData.queryPriority === undefined) {
```
Comment (Tamirk) [inline src/config_ctrl.ts:40]: Why not do `!this.current.jsonData.queryPriority` to also make sure empty value or something else

File: src/datasource.ts [hunk 2]
```diff
@@ -566,35 +570,44 @@ export class BigQueryDatasource {
       });
   }
 
-  private async doQueryRequest(query, requestId, maxRetries = 3) {
-    const path = `v2/projects/${this.runInProject}/queries`;
+  private async doQueryRequest(query, requestId, priority, maxRetries = 3) {
+    const location = this.queryModel.target.location || this.processingLocation || "US";
+    let data, queryiesOrJobs;
+    if(priority === 'INTERACTIVE'){
```
Comment (Tamirk) [inline src/datasource.ts:576]: Are you sure `priority` is always upper-case?

**Comment by Tamirk** [comment]
Can we show our client how to use it before Grafana approve the plug-in?

**Comment by Tamirk** [comment]
@LiorRacer why do we have failed message from codecov?

File: src/datasource.ts (hunk 2 above)
Comment (LiorRacer) [reply src/datasource.ts:576]: It is coming from hard coded list, but it is a good practice to match the cases

File: src/config_ctrl.ts (hunk 1 above)
Comment (LiorRacer) [reply src/config_ctrl.ts:40]: It is coming from hard coded list,

**Comment by LiorRacer** [comment]
> @LiorRacer Can we show our client how to use it before Grafana approve the plug-in?

Of course

**Pull request merged**